src/analyses/goto_check.cpp

```diff
@@ -993,9 +992,10 @@ void goto_checkt::pointer_validity_check(
   const dereference_exprt &expr,
   const guardt &guard,
   const exprt &access_lb,
-  const exprt &access_ub)
+  const exprt &access_ub,
+  const irep_idt &mode)
 {
-  if(!enable_pointer_check)
+  if(mode!=ID_java && !enable_pointer_check)
```

Comment:
Ok; I'd thus suggestion that the help output for --pointer-check be amended to say "C/C++ only".